pkg/apis/core/validation/validation.go

```diff
@@ -5974,6 +5977,91 @@ func validateWindowsSecurityContextOptions(windowsOptions *core.WindowsSecurityC
 	return allErrs
 }
 
+func validateWindowsHostProcessPod(podSpec *core.PodSpec, fieldPath *field.Path, opts PodValidationOptions) field.ErrorList {
```

Comment:
I don't think addition of ephemeral containers to a pod go through this validation path. Before this or ephemeral containers graduates from alpha, we need to ensure ephemeralcontainers are also checked for enablement of hostProcess and consistency of effective hostProcess.

cc @verb

can both of you add a tracking item to your alpha graduation criteria so this doesn't get dropped?

Reply:
Interesting. Agreed, it would be safer to validate the entire PodSpec after updating the ephemeral containers.

Reply:
https://github.com/kubernetes/enhancements/pull/2748